server/src/main/java/io/crate/metadata/upgrade/IndexTemplateUpgrader.java

```diff
@@ -170,4 +178,79 @@ private static void populateColumnPositions(String parentName,
                                     maxColumnPosition);
         }
     }
+
+    // bug : https://github.com/crate/crate/issues/13380
+    private static Map<String, IndexTemplateMetadata> fixCorruptedTemplatesFromSwapTableBug(Map<String, IndexTemplateMetadata> templates) {
+        Map<String, IndexTemplateMetadata> fixedTemplates = new HashMap<>();
+        for (Map.Entry<String, IndexTemplateMetadata> e : templates.entrySet()) {
+            RelationName corruptedRelationName = identifyCorruptedTemplateNames(e.getKey());
+            if (corruptedRelationName != null) {
+                String partnerTemplateName = PartitionName.templateName(corruptedRelationName.schema(), corruptedRelationName.name());
+                IndexTemplateMetadata corruptedTemplate = e.getValue();
+                IndexTemplateMetadata partnerTemplate = templates.get(partnerTemplateName);
+                if (partnerTemplate == null) {
+                    partnerTemplate = fixedTemplates.remove(partnerTemplateName);
```

Comment:
Looks like this can be removed as it's a double `partnerTemplate == null` check? also I don't understand why it should be removed if it will be added later on anyway...

Reply:
Thanks!